src/Shared/DomainEventWrapperCollection.cs

```diff
@@ -19,12 +20,14 @@ private DomainEventWrapperCollection(IEnumerable<DomainEventWrapper> domainEvent
 
     public EventSchema DomainSchema { get; }
 
-    public static DomainEventWrapperCollection Create<T>(IEnumerable<T> domainEvents)
+    public Action<IDomainEventMetadata>? DomainEventConfiguration { get; }
```

Comment:
Why not name the property `DomainEventMetadata`?

Reply:
I am guessing you meant why not `DomainEventMetadataConfiguration` instead of simply `DomainEventConfiguration`? I followed the parameter naming, and I think I must have skipped that part when naming the parameter, so it entirely escaped my mind afterwards.

Although to be more consistent with existing APIs (such as `IServiceCollection.Configure()`) a more apt name for the parameter would be `configureDomainEventMetadata`, leading to this property being named something like `ConfigureDomainEventMetadata`.

Would that be acceptable?